dev/service-registry/client-javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/registry/ServiceRegistryServiceLocator.scala

```diff
@@ -26,9 +26,15 @@ private[lagom] class ServiceRegistryServiceLocator @Inject() (
 ) extends CircuitBreakingServiceLocator(circuitBreakers) {
 
   override def locateAll(name: String, serviceCall: Call[_, _]): CompletionStage[JList[URI]] =
-    client.locateAll(name).map(_.asJava).toJava
+    // a ServiceLocator doesn't know what a `portName` is so we fallback to `None` and the
+    // implementation will return any registry without a name. For compatibility reasons,
+    // any service regsitered using `http` as portName will also be registered without name.
+    client.locateAll(name, None).map(_.asJava).toJava
 
   override def locate(name: String, serviceCall: Call[_, _]): CompletionStage[Optional[URI]] =
-    client.locateAll(name).map(_.headOption.asJava).toJava
+    // a ServiceLocator doesn't know what a `portName` is so we fallback to `None` and the
+    // implementation will return any registry without a name. For compatibility reasons,
+    // any service regsitered using `http` as portName will also be registered without name.
+    client.locateAll(name, None).map(_.headOption.asJava).toJava
```

Comment:
I'm not 100% sure of the statement in the comments. I will have to review the behavior in Akka's Service Descriptor to make sure what the expected behavior is when a `Lookup("inventory", None, None)` is sent.

NOTE:  `Lookup("inventory", None, None)` is a lookup operation where the client doesn't care about the protocol or the portname. If `"invetory"` had two registries (one for `"http"` and another for `"https"`) I'm not sure what the expected result of the lookup should be.

In the implementation on this PR, when a `Lookup("inventory", None, None)` happens and there's more than one item on the list we return the one named `http`.